### PR TITLE
Added spacing option for writing to JSON.

### PR DESCRIPTION
**Pull request by Unknown025**
Adds support for JSON pretty printing, or passing other spacing parameters to JavaScript's JSON.stringify method.

### CODE REVIEW

File: lib/storage.js [hunk 1]
```diff
@@ -285,6 +285,7 @@ exports.getAll = function(options, callback) {
  * @param {Object} [options] - options
  * @param {String} [options.dataPath] - data path
  * @param {String} [options.validate] - validate writes by reading the data back
+ * @param {number} [options.spacing] - spacing passed to JSON.stringify
```
Comment (jviotti) [inline lib/storage.js:288]: Can we make it a `prettyprint` boolean option instead and default to 2 spaces. I don't think people are interested in the particular amount of spaces to use, but just in whether it is pretty printed or not.

Comment (Unknown025) [reply lib/storage.js]: Sorry, I forgot to reply. I updated the pull request to replace it with a boolean option.

**Comment by jviotti** [comment]
Looks good. Thanks a lot!

**Comment by jviotti** [comment]
Hey @Unknown025, I'll be releasing this as v4.3.0. It should be available in NPM today. Thanks!

**Comment by jviotti** [comment]
Published!

**Comment by Unknown025** [comment]
Awesome, thanks!

**Pull request merged**